core/arch/x86_64/kernel/thread_64.S

```diff
@@ -49,32 +49,23 @@
     popfq
 .endm
 
-/* void thread_resume(struct thread_ctx_regs *regs, struct thread_smc_args *args,
- *                      vaddr_t tmp_stack)
+/* void thread_resume(vaddr_t sp, struct thread_smc_args *args,
```

Comment:
Can we just keep one parameter struct thread_ctx_regs *regs for thread_resume? Let's try to get some ideas from Arm code.

Reply:
No. thread_ctx_regs  is only for registers. In arm design, thread_resume never returns. But for x86, it needs struct thread_smc_args *args for return values. x86 also needs tmp_stack to store current stack.